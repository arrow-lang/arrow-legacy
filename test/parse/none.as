let x: None; // type
None; // expression
