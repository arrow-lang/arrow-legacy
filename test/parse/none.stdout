{
  "tag": "Module",
  "span": "test/parse/none.as:1.1-2.5",
  "name": "none",
  "block": {
    "tag": "Block",
    "span": "test/parse/none.as:1.1-2.5",
    "statements": [
      {
        "tag": "Slot",
        "span": "test/parse/none.as:1.1-13",
        "exported": false,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/none.as:1.5-6",
          "text": "x",
          "mutable": false
        },
        "type": {
          "tag": "TypeNone",
          "span": "test/parse/none.as:1.8-12"
        },
        "initializer": null
      },
      {
        "tag": "None",
        "span": "test/parse/none.as:2.1-5"
      }
    ]
  }
}
